client/src/components/Form/Elements/FormSelect.vue

```diff
@@ -51,19 +55,22 @@ const emit = defineEmits<{
     (e: "input", value: SelectValue | Array<SelectValue>): void;
 }>();
 
+const filter = ref("");
+const filteredOptions = useFilterObjectArray(props.options, filter, ["label", "value"]);
```

Comment:
@ElectronicBlueberry is this what you had in mind?

Unfortunately, I cannot filter by tags because the filter expects only the keys of the object and in this case, the tags are in the sub-object "value.tags" :thinking:

Reply:
I decided to allow referencing nested fields when filtering with a "naive" approach of providing a list of strings as "path" to the nested field in https://github.com/galaxyproject/galaxy/pull/18918/commits/5bccb6f28d2dc7c8133afa8c346dcd7ed9cd356f

It works on my testing although I had to add a mock implementation because obviously, the existing jest tests didn't like the worker :sweat_smile:

Reply:
Hmmm nope, now there are some issues when switching histories... when you try to filter after switching histories it doesn't show the correct items... I must have missed something... :thinking: I'm probably using the worker in a suboptimal way since I can see some warnings in the console too:

```
[Vue warn]: onScopeDispose() is called when there is no active effect scope to be associated with.
```

Any ideas?

Reply:
The options are not updating because passing props.options to the composable is not reactive. This is because the acessor on props is called in the components setup function during the function call, and not within the composable, so the reactive dependency cannot be established. Wrap props.options in a computed or getter function to fix this. This can be done inline.

Reply:
Do you mean something like https://github.com/galaxyproject/galaxy/pull/18918/commits/72bb11c10c359385707535fdec47e8fcfbc75b26?

I'm afraid it doesn't work or I'm not getting it right... :thinking:

Reply:
Thank you @ElectronicBlueberry I think I got it working now! :+1: